README.md
# es_freeroam v0.1.2
<a href="https://discord.gg/eNJraMf"><img alt="Discord Status" src="https://discordapp.com/api/guilds/285462938691567627/widget.png"></a>

es_freeroam is a FiveM game mode with a money system.  
The player can receive jobs, survivals, buy buildings, drugs and many more.

 **Note: This project is at an early development stage and we will only continue if other developers contribute to it.   
 you can find the current changelog [here](CHANGELOG.MD)**

## Requirements

- [Essentialmode](https://forum.fivem.net/t/release-essentialmode-base/3665)

## Installation

1. [Download](https://github.com/FiveM-Scripts/Essential_Freeroam/archive/master.zip)
2. Extract the folder and rename it to es_freeroam
3. Place the folder in your resources/[essential] folder
4. Add - es_freeroam to your AutoStartResources in citmp-server.yml
5. Change **resource_type 'map' { gameTypes = { fivem = true } }** to **resource_type 'map' { gameTypes = { essentialmode = true } }** in your fivem-map-skater or fivem-map-hipster **resource.lua** file.
6. Restart your server

## Contribute
if you are a developer and  would like to contribute any help is welcome!   
The contribution guide can be found [here](CONTRIBUTING.MD).


CHANGELOG.MD
# Changelog
The information about the changes can be found below.   

## v0.1.2
#### Features
- Added "Fame or Shame" (Maze Bank Arena) IPL and blip
- Players can now buy weed at "Smoke on The water"
- Added the jewelry store IPL and blip
- Added the stripclub
- Added /rmwanted to remove your wanted level
- Added /help command

#### Fixes
- Added Lester's factory missing IPL.
- Added Lesters factory to the map.
- The scoreboard now shows the ID for each player

## v0.1.1 (23-3-2017)   
#### Features
- Added bilips on the map for stores, police stations, survivals, and many more...
- Added missing IPLS.

#### Fixes
- Fixed compatibility with essential mode 2.0   

## v0.1.0 (27-2-2017)  
#### Features
- Added money integration with essential mode

#### Fixes
- none


client.lua
-- Spawn override
AddEventHandler('onClientMapStart', function()
    exports.spawnmanager:setAutoSpawn(true)
    exports.spawnmanager:forceRespawn()
end)

-- Allows the server to spawn the player
RegisterNetEvent('es_freeroam:spawnPlayer')
AddEventHandler('es_freeroam:spawnPlayer', function(x, y, z, model)
    exports.spawnmanager:spawnPlayer({x = x, y = y, z = z, model = model})
end)

  AddEventHandler("playerSpawned", function(spawn)
    -- Send notifications
    Citizen.CreateThread(function()
    while true do
      Wait(0)
      DisplayCash(false);
      SetNotificationTextEntry("STRING");
      AddTextComponentString("Welcome to ~g~FiveM!.\n ~y~For more info go to github.com/FiveM-Scripts");
      SetNotificationMessage("CHAR_ALL_PLAYERS_CONF", "CHAR_ALL_PLAYERS_CONF", true, 1, "Essential Freeroam", "v0.1.2");
      DrawNotification(false, true);
      Wait(5000000)
    end
 end)

   -- give the player some weapons
   Citizen.CreateThread(function()
     GiveWeaponToPed(GetPlayerPed(-1), GetHashKey("WEAPON_KNIFE"), true, true)
   end)
 end)

 RegisterNetEvent("es_freeroam:wanted")
 AddEventHandler("es_freeroam:wanted", function()
   Citizen.CreateThread(function()
       SetPlayerWantedLevel(PlayerId(), 0, 0)
       SetPlayerWantedLevelNow(PlayerId(), 0)
     end)
 end)

  -- Display text
    RegisterNetEvent("es_freeroam:displaytext")
    AddEventHandler("es_freeroam:displaytext", function(text, time)
      ClearPrints()
    	SetTextEntry_2("STRING")
    	AddTextComponentString(text)
    	DrawSubtitleTimed(time, 1)
    end)

    -- Display notification
    RegisterNetEvent("es_freeroam:notify")
    AddEventHandler("es_freeroam:notify", function(icon, type, sender, title, text)
      Citizen.CreateThread(function()
        Wait(1)
        SetNotificationTextEntry("STRING");
        AddTextComponentString(text);
        SetNotificationMessage(icon, icon, true, type, sender, title, text);
        DrawNotification(false, true);
      end)
    end)


server.lua
RegisterServerEvent('playerSpawn')
AddEventHandler('playerSpawn', function()
	TriggerClientEvent('es_freeroam:spawnPlayer', source, 464.091, -997.166, 24.915)
end)

AddEventHandler('es:playerLoaded', function(source)
	-- Get the players money amount
	TriggerEvent("es:getPlayerFromId", source, function(user)
	user:setMoney((user.money))
	end)
end)

RegisterServerEvent('mission:completed')
AddEventHandler('mission:completed', function(total)
	-- Get the players money amount
TriggerEvent('es:getPlayerFromId', source, function(user)
  -- update player money amount
  user:addMoney((total))
 TriggerClientEvent("es_freeroam:notify", source, "CHAR_BANK_MAZE", 1, "Maze Bank", false, "You received ~g~$".. tonumber(total))
 end)
end)

RegisterServerEvent('es_freeroam:pay')
AddEventHandler('es_freeroam:pay', function(amount)
	-- Get the players money amount
	TriggerEvent("es:getPlayerFromId", source, function(user)
		if(user.money > amount) then
			TriggerClientEvent("es_freeroam:notify", source, "CHAR_BANK_MAZE", 1, "Maze Bank", false, "Your transaction is ~g~completed.")
			user:removeMoney((amount))
		else
			TriggerClientEvent("es_freeroam:notify", source, "CHAR_BANK_MAZE", 1, "Maze Bank", false, "Your transaction is ~r~rejected.")
		end
	end)
end)


events/smoke.lua
local thisPed
local pedCoords = {}
local storedPeds = {}
local blips = {
  -- Smoke on the water
  {x=-1171.42, y=-1572.72, z=3.6636},
}

local Keys = {
	["ESC"] = 322, ["F1"] = 288, ["F2"] = 289, ["F3"] = 170, ["F5"] = 166, ["F6"] = 167, ["F7"] = 168, ["F8"] = 169, ["F9"] = 56, ["F10"] = 57,
	["~"] = 243, ["1"] = 157, ["2"] = 158, ["3"] = 160, ["4"] = 164, ["5"] = 165, ["6"] = 159, ["7"] = 161, ["8"] = 162, ["9"] = 163, ["-"] = 84, ["="] = 83, ["BACKSPACE"] = 177,
	["TAB"] = 37, ["Q"] = 44, ["W"] = 32, ["E"] = 38, ["R"] = 45, ["T"] = 245, ["Y"] = 246, ["U"] = 303, ["P"] = 199, ["["] = 39, ["]"] = 40, ["ENTER"] = 18,
	["CAPS"] = 137, ["A"] = 34, ["S"] = 8, ["D"] = 9, ["F"] = 23, ["G"] = 47, ["H"] = 74, ["K"] = 311, ["L"] = 182,
	["LEFTSHIFT"] = 21, ["Z"] = 20, ["X"] = 73, ["C"] = 26, ["V"] = 0, ["B"] = 29, ["N"] = 249, ["M"] = 244, [","] = 82, ["."] = 81,
	["LEFTCTRL"] = 36, ["LEFTALT"] = 19, ["SPACE"] = 22, ["RIGHTCTRL"] = 70,
	["HOME"] = 213, ["PAGEUP"] = 10, ["PAGEDOWN"] = 11, ["DELETE"] = 178,
	["LEFT"] = 174, ["RIGHT"] = 175, ["TOP"] = 27, ["DOWN"] = 173,
	["NENTER"] = 201, ["N4"] = 108, ["N5"] = 60, ["N6"] = 107, ["N+"] = 96, ["N-"] = 97, ["N7"] = 117, ["N8"] = 61, ["N9"] = 118
}

local MISSION = {}
MISSION.start = false
MISSION.wanted = false
local playerCoords
local playerPed

showStartText   = false

--blips
local BLIP = {}
Citizen.CreateThread(function()
    while true do
       Wait(0)
       playerPed = GetPlayerPed(-1)
       playerCoords = GetEntityCoords(playerPed, 0)
      tick()
    end
end)

function tick()
    --Show notification, when player is near the weedshop
    if(MISSION.start == false) then
    for _, item in pairs(blips) do
    if(GetDistanceBetweenCoords(playerCoords, item.x, item.y, item.z) < 10) then
            if(showStartText == false) then
                StartText()
            end
                -- Start mission
                if(IsControlPressed(1, Keys["E"])) then
                  TriggerServerEvent("es_freeroam:pay", tonumber(50))
                    Toxicated()
                    Citizen.Wait(120000)
                    reality()
                end
              else
                showStartText = false
              end --if GetDistanceBetweenCoords ...
            end -- end for
          end--if MISSION.start == false


                    if(MISSION.start == true) then
                      Citizen.CreateThread(function()
                        while true do
                          vehCheck = IsPedInAnyVehicle(GetPlayerPed(-1), true)
                            if vehCheck then
                              SetPlayerWantedLevel(GetPlayerPed(-1), 1, 0)
                              SetPlayerWantedLevelNow(GetPlayerPed(-1), 0)
                              Citizen.Wait(59000)
                            end
                        end
                      end)
                    end -- end mission.start
                  end -- end tick

function Toxicated()
  TaskStartScenarioInPlace(GetPlayerPed(-1), "WORLD_HUMAN_DRUG_DEALER", 0, 1)
  DoScreenFadeOut(1000)
  Citizen.Wait(1000)
  ClearPedTasksImmediately(GetPlayerPed(-1))
  SetTimecycleModifier("spectator5")
  SetPedMotionBlur(GetPlayerPed(-1), true)
  RequestAnimSet("MOVE_M@DRUNK@SLIGHTLYDRUNK")
      while not HasAnimSetLoaded("MOVE_M@DRUNK@SLIGHTLYDRUNK") do
       Citizen.Wait(0)
      end
    SetPedMovementClipset(GetPlayerPed(-1), "MOVE_M@DRUNK@SLIGHTLYDRUNK", true)
    SetPedIsDrunk(GetPlayerPed(-1), true)
    DoScreenFadeIn(1000)
      MISSION.start = true
      showStartText = true
  end

  function reality()
    Citizen.Wait(50000)
    DoScreenFadeOut(1000)
    Citizen.Wait(1000)
    DoScreenFadeIn(1000)
    ClearTimecycleModifier()
    ResetScenarioTypesEnabled()
    ResetPedMovementClipset(GetPlayerPed(-1), 0)
    SetPedIsDrunk(GetPlayerPed(-1), false)
    SetPedMotionBlur(GetPlayerPed(-1), false)
    -- Disable wanted level
    SetPlayerWantedLevel(GetPlayerPed(-1), 0, 0)
    SetPlayerWantedLevelNow(GetPlayerPed(-1), 0)
    -- Stop the mini mission
    MISSION.start = false
    Citizen.Trace("Going back to reality\n")
    end

function StartText()
  DrawMarker(1, -1171.42, -1572.72, 3.6636, 0, 0, 0, 0, 0, 0, 4.0, 4.0, 2.0, 178, 236, 93, 155, 0, 0, 2, 0, 0, 0, 0)
  ShowInfo("Press ~INPUT_CONTEXT~ to buy drugs", 0)
end

function drawTxt(x,y ,width,height,scale, text, r,g,b,a)
    SetTextFont(0)
    SetTextProportional(0)
    SetTextScale(scale, scale)
    SetTextColour(r, g, b, a)
    SetTextDropShadow(0, 0, 0, 0,255)
    SetTextEdge(1, 0, 0, 0, 255)
    SetTextDropShadow()
    SetTextOutline()
    SetTextEntry("STRING")
    AddTextComponentString(text)
    DrawText(x - width/2, y - height/2 + 0.005)
end

function DrawMissionText(m_text, showtime)
  ClearPrints()
  SetTextEntry_2("STRING")
  AddTextComponentString(m_text)
  DrawSubtitleTimed(showtime, 1)
end


function ShowNotification(text)
    SetNotificationTextEntry("STRING")
    AddTextComponentString(text)
    DrawNotification(true, false)
end

function ShowInfo(text, state)
	SetTextComponentFormat("STRING")
	AddTextComponentString(text)
	DisplayHelpTextFromStringLabel(0, state, 0, -1)
end


stores/stripclub.lua
-- Configure the coordinates where the strippers should be placed.
local strippers = {
  {type=5, hash=0x2970a494, x=112.159, y=-1287.326, z=28.459, a=265.902},
  {type=5, hash=0x2970a494, x=108.440, y=-1289.298, z=28.859, a=338.700},
  {type=5, hash=0x2970a494, x=108.181, y=-1304.807, z=28.769, a=186.893},
  {type=5, hash=0x2970a494, x=118.125, y=-1283.357, z=28.277, a=124.466},
  }

-- Configure the coordinates for the bartenders.
  local bartenders = {
    {type=5, hash=0x780c01bd, x=128.900, y=-1283.211, z=29.273, a=123.98},
  }

-- Configure the coordinates for the bartenders.
local bouncers = {
  {type=4, hash=0x9fd4292d, x=130.328, y=-1298.409, z=29.233, a=211.486},
  {type=4, hash=0x9fd4292d, x=127.404, y=-1300.126, z=29.23, a=211.587},
  {type=4, hash=0x9fd4292d, x=111.088, y=-1304.371, z=29.020, a=296.699},
}

function LocalPed()
  return GetPlayerPed(-1)
end

Citizen.CreateThread(function()
  -- Load the ped modal (s_f_y_bartender_01)
  RequestModel(GetHashKey("s_f_y_bartender_01"))
  while not HasModelLoaded(GetHashKey("s_f_y_bartender_01")) do
    Wait(1)
  end

  -- Load the ped modal (mp_f_stripperlite)
  RequestModel(GetHashKey("mp_f_stripperlite"))
  while not HasModelLoaded(GetHashKey("mp_f_stripperlite")) do
    Wait(1)
  end

  -- Load the ped modal (s_m_m_bouncer_01)
  RequestModel(GetHashKey("s_m_m_bouncer_01"))
  while not HasModelLoaded(GetHashKey("s_m_m_bouncer_01")) do
    Wait(1)
  end

  -- Load the animation (testing)
  RequestAnimDict("mini@strip_club@idles@stripper")
  while not HasAnimDictLoaded("mini@strip_club@idles@stripper") do
    Wait(1)
  end

  -- Load the bouncer animation (testing)
  RequestAnimDict("mini@strip_club@idles@bouncer@base")
  while not HasAnimDictLoaded("mini@strip_club@idles@bouncer@base") do
    Wait(1)
  end

    -- Spawn the bartender to the coordinates
    bartender =  CreatePed(5, 0x780c01bd, 128.900, -1283.21, 29.273, 123.98, false, true)
    SetBlockingOfNonTemporaryEvents(bartender, true)
    SetPedCombatAttributes(bartender, 46, true)
    SetPedFleeAttributes(bartender, 0, 0)
    SetPedRelationshipGroupHash(bartender, GetHashKey("CIVFEMALE"))


  -- Spawn the bouncers to the coordinates
  for _, item in pairs(bouncers) do
    ped =  CreatePed(item.type, item.hash, item.x, item.y, item.z, item.a, false, true)
    GiveWeaponToPed(ped, 0x1B06D571, 2800, false, true)
    SetPedCombatAttributes(ped, 46, true)
    SetPedFleeAttributes(ped, 0, 0)
    SetPedArmour(ped, 100)
    SetPedMaxHealth(ped, 100)
    SetPedRelationshipGroupHash(ped, GetHashKey("army"))
    TaskStartScenarioInPlace(ped, "WORLD_HUMAN_GUARD_STAND_PATROL", 0, true)
    SetPedCanRagdoll(ped, false)
    SetPedDiesWhenInjured(ped, false)
    TaskPlayAnim(ped,"mini@strip_club@idles@bouncer@base","base", 8.0, 0.0, -1, 1, 0, 0, 0, 0)
    end

  -- Spawn the strippers to the coordinates
  for _, item in pairs(strippers) do
    stripper =  CreatePed(item.type, item.hash, item.x, item.y, item.z, item.a, false, true)
    GiveWeaponToPed(stripper, 0x99B507EA, 2800, false, true)
    SetPedCombatAttributes(stripper, 46, true)
    SetPedFleeAttributes(stripper, 0, 0)
    SetPedArmour(stripper, 200)
    SetPedMaxHealth(stripper, 200)
    SetPedDiesWhenInjured(ped, false)
    SetPedRelationshipGroupHash(stripper, GetHashKey("army"))
    TaskPlayAnim(stripper,"mini@strip_club@idles@stripper","stripper_idle_03", 8.0, 0.0, -1, 1, 0, 0, 0, 0)
  end

end)


player/scoreboard.lua
local Settings = {
	--Should the scoreboard draw wanted level?
	["WantedStars"] = false,

	--Should the scoreboard draw player ID?
	["PlayerID"] = true,

	--Should the scoreboard draw voice indicator?
	["VoiceIndicator"] = true,

	--Display time in milliseconds
	["DisplayTime"] = 5000,

	--Keys that will activate the scoreboard.
	--Change only the number value, NOT the 'true'
	--Multiple keys can be used, simply add another row with another number
	--See: https://wiki.gtanet.work/index.php?title=Game_Controls

	--MultiplayerInfo / Z
	[20] = true,

	--Detonate / G
	--[47] = true,
}

-- END OF SETTINGS --

local function DrawPlayerList()
    local players = {}

    for i = 0, 31 do
        if NetworkIsPlayerActive( i ) then
            table.insert( players, i )
        end
    end

	--Top bar
	DrawRect( 0.11, 0.025, 0.2, 0.03, 0, 0, 0, 220 )

	--Top bar title
	SetTextFont( 4 )
	SetTextProportional( 0 )
	SetTextScale( 0.45, 0.45 )
	SetTextColour( 255, 255, 255, 255 )
	SetTextDropShadow( 0, 0, 0, 0, 255 )
	SetTextEdge( 1, 0, 0, 0, 255 )
	SetTextEntry( "STRING" )
	AddTextComponentString( "Players: " .. #players )
	DrawText( 0.015, 0.007 )

	for k, v in pairs( players ) do
		local r
		local g
		local b

		if k % 2 == 0 then
			r = 28
			g = 47
			b = 68
		else
			r = 38
			g = 57
			b = 74
		end

		--Row BG
		DrawRect( 0.11, 0.025 + ( k * 0.03 ), 0.2, 0.03, r, g, b, 220 )

		--Name Label
		SetTextFont( 4 )
		SetTextScale( 0.45, 0.45 )
		SetTextColour( 255, 255, 255, 255 )
		SetTextEntry( "STRING" )
		AddTextComponentString( GetPlayerName( v ) )
		DrawText( 0.015, 0.007 + ( k * 0.03 ) )


		--Voice Indicator
		if Settings["VoiceIndicator"] then
			local transparency = 60

			if NetworkIsPlayerTalking( v ) then
				transparency = 255
			end

			DrawSprite( "mplobby", "mp_charcard_stats_icons9", 0.2, 0.024 + ( k * 0.03 ), 0.015, 0.025, 0, 255, 255, 255, transparency )
		end

		--Wanted Stars
		if Settings["WantedStars"] then
			local wantedLevel = GetPlayerWantedLevel( v ) or 0
			wantedLevel = wantedLevel

			for i=1, 5 do
				local transparency = 60

				if wantedLevel >= i then
					transparency = 255
				end

				DrawSprite( "mpleaderboard", "leaderboard_star_icon", 0.195 - ( i * 0.010	 ), 0.024 + ( k * 0.03 ), 0.0175, 0.0275, 0, 255, 255, 255, transparency )
			end
		end
	end
end

local LastPress = 0

Citizen.CreateThread( function()
	RequestStreamedTextureDict( "mplobby" )
	RequestStreamedTextureDict( "mpleaderboard" )

	while true do
		Wait( 0 )

		for k, v in pairs( Settings ) do
			if type( k ) == "number" and v == true then
				if IsControlPressed( 0, k ) then
					LastPress = GetGameTimer()
				end
			end
		end

		if GetGameTimer() < LastPress + Settings["DisplayTime"] then
			DrawPlayerList()
		end

	end
end )


player/map.lua
Citizen.CreateThread(function()

  local blips = {
     -- Airport and Airfield
     {name="Airport", id=90, x=-1032.690, y=-2728.141, z=13.757},
     {name="Airport", id=90, x=1743.6820, y=3286.2510, z=40.087},
     -- barbers
     {name="Barber", id=71, x=-827.333, y=-190.916, z=37.599},
     {name="Barber", id=71, x=130.512, y=-1715.535, z=29.226},
     {name="Barber", id=71, x=-1291.472, y=-1117.230, z=6.641},
     {name="Barber", id=71, x=1936.451, y=3720.533, z=32.638},
     {name="Barber", id=71, x=1200.214, y=-468.822, z=66.268},
     {name="Barber", id=71, x=-30.109, y=-141.693, z=57.041},
     {name="Barber", id=71, x=-285.238, y=6236.365, z=31.455},
     -- Stores
      {name="Store", id=52, x=28.463, y=-1353.033, z=29.340},
      {name="Store", id=52, x=-54.937, y=-1759.108, z=29.005},
      {name="Store", id=52, x=375.858, y=320.097, z=103.433},
      {name="Store", id=52, x=1143.813, y=-980.601, z=46.205},
      {name="Store", id=52, x=1695.284, y=4932.052, z=42.078},
      {name="Store", id=52, x=2686.051, y=3281.089, z=55.241},
      {name="Store", id=52, x=1967.648, y=3735.871, z=32.221},
      {name="Store", id=52, x=-2977.137, y=390.652, z=15.024},
      {name="Store", id=52, x=1160.269, y=-333.137, z=68.783},
      {name="Store", id=52, x=-1492.784, y=-386.306, z=39.798},
      {name="Store", id=52, x=-1229.355, y=-899.230, z=12.263},
      {name="Store", id=52, x=-712.091, y=-923.820, z=19.014},
      {name="Store", id=52, x=-1816.544, y=782.072, z=137.600},
      {name="Store", id=52, x=1729.689, y=6405.970, z=34.453},
      {name="Store", id=52, x=2565.705, y=385.228, z=108.463},
      -- Clothing
      {name="Clothing", id=73, x=88.291, y=-1391.929, z=29.200},
      {name="Clothing", id=73, x=-718.985, y=-158.059, z=36.996},
      {name="Clothing", id=73, x=-151.204, y=-306.837, z=38.724},
      {name="Clothing", id=73, x=414.646, y=-807.452, z=29.338},
      {name="Clothing", id=73, x=-815.193, y=-1083.333, z=11.022},
      {name="Clothing", id=73, x=-1208.098, y=-782.020, z=17.163},
      {name="Clothing", id=73, x=-1457.954, y=-229.426, z=49.185},
      {name="Clothing", id=73, x=-2.777, y=6518.491, z=31.533},
      {name="Clothing", id=73, x=1681.586, y=4820.133, z=42.046},
      {name="Clothing", id=73, x=130.216, y=-202.940, z=54.505},
      {name="Clothing", id=73, x=618.701, y=2740.564, z=41.905},
      {name="Clothing", id=73, x=1199.169, y=2694.895, z=37.866},
      {name="Clothing", id=73, x=-3164.172, y=1063.927, z=20.674},
      {name="Clothing", id=73, x=-1091.373, y=2702.356, z=19.422},
      -- ammunationblips
      {name="Weapon store", id=110, x=1701.292, y=3750.450, z=34.365},
      {name="Weapon store", id=110, x=237.428, y=-43.655, z=69.698},
      {name="Weapon store", id=110, x=843.604, y=-1017.784, z=27.546},
      {name="Weapon store", id=110, x=-321.524, y=6072.479, z=31.299},
      {name="Weapon store", id=110, x=-664.218, y=-950.097, z=21.509},
      {name="Weapon store", id=110, x=-1320.983, y=-389.260, z=36.483},
      {name="Weapon store", id=110, x=-1109.053, y=2686.300, z=18.775},
      {name="Weapon store", id=110, x=2568.379, y=309.629, z=108.461},
      {name="Weapon store", id=110, x=-3157.450, y=1079.633, z=20.692},
      -- Basic
      {name="Comedy Club", id=102, x=377.088, y=-991.869, z=-97.604},
      {name="Franklin", id=210, x=7.900, y=548.100, z=175.500},
      {name="Franklin", id=210, x=-14.128,	y=-1445.483,	z=30.648},
      {name="Michael", id=124, x=-852.400, y=160.000, z=65.600},
      {name="Trevor", id=208, x=1985.700, y=3812.200, z=32.200},
      {name="Trevor", id=208, x=-1159.034,	y=-1521.180, z=10.633},
      {name="FIB", id=106, x=105.455, y=-745.483, z=44.754},
      {name="Lifeinvader", id=77, x=-1047.900, y=-233.000, z=39.000},
      {name="Cluckin Bell", id=357, x=-72.68752, y=6253.72656, z=31.08991},
      {name="Tequil-La La", id=93, x=-565.171, y=276.625, z=83.286},
      {name="O'Neil Ranch", id=438, x=2441.200, y=4968.500, z=51.700},
      {name="Play Boy Mansion", id=439, x=-1475.234, y=167.088, z=55.841},
      {name="Hippy Camp", id=140, x=2476.712, y=3789.645, z=41.226},
      {name="Chop shop", id=446, x=479.056, y=-1316.825, z=28.203},
      {name="Rebel Radio", id=136, x=736.153, y=2583.143, z=79.634},
      {name="Morgue", id=310, x=243.351, y=-1376.014, z=39.534},
      {name="Golf", id=109, x=-1336.715, y=59.051, z=55.246 },
      -- Propperty
      {name="Casino", id=207, x=925.329, y=46.152, z=80.908 },
      {name="Maze Bank Arena", id=135, x=-250.604, y=-2030.000, z=30.000},
      {name="Stripbar", id=121, x=134.476, y=-1307.887, z=28.983},
      {name="Smoke on the Water", id=140, x=-1171.42, y=-1572.72, z=3.6636},
      {name="Weed Farm", id=140, x=2208.777, y=5578.235, z=53.735},
      {name="Downtown Cab Co", id=375, x=900.461, y=-181.466, z=73.89},
      {name="Theater", id=135, x=293.089, y=180.466, z=104.301},
      -- Gangs
      {name="Gang", id=437, x=298.68, y=-2010.10, z=20.07},
      {name="Gang", id=437, x=86.64, y=-1924.60, z=20.79},
      {name="Gang", id=437, x=-183.52, y=-1632.62, z=33.34},
      {name="Gang", id=437, x=989.37, y=-1777.56, z=31.32},
      {name="Gang", id=437, x=960.24, y=-140.31, z=74.50},
      {name="Gang", id=437, x=-1042.29, y=4910.17, z=94.92},
      -- Gas stations
      {name="Gas Station", id=361, x=49.4187,   y=2778.793,  z=58.043},
      {name="Gas Station", id=361, x=263.894,   y=2606.463,  z=44.983},
      {name="Gas Station", id=361, x=1039.958,  y=2671.134,  z=39.550},
      {name="Gas Station", id=361, x=1207.260,  y=2660.175,  z=37.899},
      {name="Gas Station", id=361, x=2539.685,  y=2594.192,  z=37.944},
      {name="Gas Station", id=361, x=2679.858,  y=3263.946,  z=55.240},
      {name="Gas Station", id=361, x=2005.055,  y=3773.887,  z=32.403},
      {name="Gas Station", id=361, x=1687.156,  y=4929.392,  z=42.078},
      {name="Gas Station", id=361, x=1701.314,  y=6416.028,  z=32.763},
      {name="Gas Station", id=361, x=179.857,   y=6602.839,  z=31.868},
      {name="Gas Station", id=361, x=-94.4619,  y=6419.594,  z=31.489},
      {name="Gas Station", id=361, x=-2554.996, y=2334.40,  z=33.078},
      {name="Gas Station", id=361, x=-1800.375, y=803.661,  z=138.651},
      {name="Gas Station", id=361, x=-1437.622, y=-276.747,  z=46.207},
      {name="Gas Station", id=361, x=-2096.243, y=-320.286,  z=13.168},
      {name="Gas Station", id=361, x=-724.619, y=-935.1631,  z=19.213},
      {name="Gas Station", id=361, x=-526.019, y=-1211.003,  z=18.184},
      {name="Gas Station", id=361, x=-70.2148, y=-1761.792,  z=29.534},
      {name="Gas Station", id=361, x=265.648,  y=-1261.309,  z=29.292},
      {name="Gas Station", id=361, x=819.653,  y=-1028.846,  z=26.403},
      {name="Gas Station", id=361, x=1208.951, y= -1402.567, z=35.224},
      {name="Gas Station", id=361, x=1181.381, y= -330.847,  z=69.316},
      {name="Gas Station", id=361, x=620.843,  y= 269.100,  z=103.089},
      {name="Gas Station", id=361, x=2581.321, y=362.039, 108.468},
      -- Police Stations
      {name="Police Station", id=60, x=425.130, y=-979.558, z=30.711},
      {name="Police Station", id=60, x=1859.234, y= 3678.742, z=33.690},
      {name="Police Station", id=60, x=-438.862, y=6020.768, z=31.490},
      {name="Police Station", id=60, x=818.221, y=-1289.883, z=26.300},
      {name="Prison", id=285, x=1679.049, y=2513.711, z=45.565},
      -- Hospitals
      {name="Hospital", id=61, x= 1839.6, y= 3672.93, z= 34.28},
      {name="Hospital", id=61, x= -247.76, y= 6331.23, z=32.43},
      {name="Hospital", id=61, x= -449.67, y= -340.83, z= 34.50},
      {name="Hospital", id=61, x= 357.43, y= -593.36, z= 28.79},
      {name="Hospital", id=61, x= 295.83, y= -1446.94, z= 29.97},
      {name="Hospital", id=61, x= -676.98, y= 310.68, z= 83.08},
      {name="Hospital", id=61, x= 1151.21, y= -1529.62, z= 35.37},
      {name="Hospital", id=61, x= -874.64, y= -307.71, z= 39.58},
      -- Vehicle Shop (Simeon)
      {name="Simeon", id=120, x=-33.803, y=-1102.322, z=25.422},
      -- LS Customs
      {name="LS Customs", id=72, x= -362.796, y= -132.400, z= 38.252},
      {name="LS Customs", id=72, x= -1140.19, y= -1985.478, z= 12.729},
      {name="LS Customs", id=72, x= 716.464, y= -1088.869, z= 21.929},
      {name="LS Customs", id=72, x= 1174.81, y= 2649.954, z= 37.371},
      {name="LS Customs", id=72, x= 118.485, y= 6619.560, z= 31.802},
      -- Lester
      {name="Lester", id=77, x=1248.183, y=-1728.104, z=56.000},
      {name="Lester", id=77, x=719.000, y=-975.000, y=25.000},
      -- Survivals
      {name="Survival", id=305, x=2351.331, y=3086.969, z=48.057},
      {name="Survival", id=305, x=-1695.803, y=-1139.190, z=13.152},
      {name="Survival", id=305, x=1532.52, y=-2138.682, z=77.120},
      {name="Survival", id=305, x=-593.724, y=5283.231, z=70.230},
      {name="Survival", id=305, x=1891.436, y=3737.409, z=32.513},
      {name="Survival", id=305, x=195.572, y=-942.493, z=30.692},
      {name="Survival", id=305, x=1488.579, y=3582.804, z=35.345},

      {name="Safehouse", id=357, x=-952.35943603516, y= -1077.5021972656, z=2.6772258281708},
      {name="Safehouse", id=357, x=-59.124889373779, y= -616.55456542969, z=37.356777191162},
      {name="Safehouse", id=357, x=-255.05390930176, y= -943.32885742188, z=31.219989776611},
      {name="Safehouse", id=357, x=-771.79888916016, y= 351.59423828125, z=87.998191833496},
      -- Yacht
      {name="Yacht", id=410, x=-2045.800, y=-1031.200, z=11.900},
      {name="Cargoship", id=410, x=-90.000, y=-2365.800, z=14.300},

    }

    for _, item in pairs(blips) do
      item.blip = AddBlipForCoord(item.x, item.y, item.z)
      SetBlipSprite(item.blip, item.id)
      SetBlipAsShortRange(item.blip, true)
      BeginTextCommandSetBlipName("STRING")
      AddTextComponentString(item.name)
      EndTextCommandSetBlipName(item.blip)
    end

  --load unloaded ipl's
  LoadMpDlcMaps()
  EnableMpDlcMaps(true)
  RequestIpl("FIBlobby")
  RemoveIpl("FIBlobbyfake")
  RequestIpl("FBI_colPLUG")
  RequestIpl("FBI_repair")
  RequestIpl("v_tunnel_hole")
  RequestIpl("TrevorsMP")
  RequestIpl("TrevorsTrailer")
  RequestIpl("TrevorsTrailerTidy")
  RemoveIpl("farm_burnt")
  RemoveIpl("farm_burnt_lod")
  RemoveIpl("farm_burnt_props")
  RemoveIpl("farmint_cap")
  RemoveIpl("farmint_cap_lod")
  RequestIpl("farm")
  RequestIpl("farmint")
  RequestIpl("farm_lod")
  RequestIpl("farm_props")
  RequestIpl("facelobby")
  RemoveIpl("CS1_02_cf_offmission")
  RequestIpl("CS1_02_cf_onmission1")
  RequestIpl("CS1_02_cf_onmission2")
  RequestIpl("CS1_02_cf_onmission3")
  RequestIpl("CS1_02_cf_onmission4")
  RequestIpl("v_rockclub")
  RemoveIpl("hei_bi_hw1_13_door")
  RequestIpl("bkr_bi_hw1_13_int")
  RequestIpl("ufo")
  RemoveIpl("v_carshowroom")
  RemoveIpl("shutter_open")
  RemoveIpl("shutter_closed")
  RemoveIpl("shr_int")
  RemoveIpl("csr_inMission")
  RequestIpl("v_carshowroom")
  RequestIpl("shr_int")
  RequestIpl("shutter_closed")
  RequestIpl("smboat")
  RequestIpl("cargoship")
  RequestIpl("railing_start")
  RemoveIpl("sp1_10_fake_interior")
  RemoveIpl("sp1_10_fake_interior_lod")
  RequestIpl("sp1_10_real_interior")
  RequestIpl("sp1_10_real_interior_lod")
  RemoveIpl("id2_14_during_door")
  RemoveIpl("id2_14_during1")
  RemoveIpl("id2_14_during2")
  RemoveIpl("id2_14_on_fire")
  RemoveIpl("id2_14_post_no_int")
  RemoveIpl("id2_14_pre_no_int")
  RemoveIpl("id2_14_during_door")
  RequestIpl("id2_14_during1")
  RequestIpl("Coroner_Int_on")
  RemoveIpl("Coroner_Int_off")
end)
